internal/service/pushbasedlogexport/model_test.go

```diff
@@ -0,0 +1,149 @@
+package pushbasedlogexport_test
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"go.mongodb.org/atlas-sdk/v20231115010/admin"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/pushbasedlogexport"
+)
+
+var (
+	testBucketName  = "test-bucket-name"
+	testIAMRoleID   = "661fe3ad234b02027ddee196"
+	testPrefixPath  = "prefix/path"
+	prefixPathEmpty = ""
+	testProjectID   = "661fe3ad234b02027dabcabc"
+)
+
+type sdkToTFModelTestCase struct {
+	apiResp         *admin.PushBasedLogExportProject
+	timeout         *timeouts.Value
+	expectedTFModel *pushbasedlogexport.TFPushBasedLogExportRSModel
+	name            string
+	projectID       string
+}
+
+func TestNewTFPushBasedLogExport(t *testing.T) {
+	currentTime := time.Now()
+
+	testCases := []sdkToTFModelTestCase{
```

Comment:
knit: i think it's a bit better using a string map for the test name, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/project/model_project_test.go#L550-L557